requests-for-comments/2022-06-10-nail-exports.md

```diff
@@ -0,0 +1,110 @@
+* API:
+  - Standard retrievel using public API
+
+
+## Method breakdown
+
+### Embeds
+| insights ✅ | dashboards ✅ | person ❌ | cohorts ❌ | groups ❌ |
+|------------|--------------|----------|-----------|----------|
+
+Similar to SharedDashboards we currently have, it makes sense to have a unique rendering method for standalone visualisations that can either be accessed:
+1. Directly via an authenticated user
+2. Directly via an unauthenticated user if "sharing" has been enabled
+3. Via an iframe embedded elsewhere
+4. Server side via selenium for taking high-quality image exports as "screenshots"
+
+---
+#### Technical considerations
+- [ ] Embeds should allow customiseable options as query params for showing hiding different elements (e.g. readonly, whitelabel)
+
+#### Current State
+- 🤔 Dashboards are embeddable and shareable under a special `shared_dashboards` endpoint
+- 🤔 Exports are _almost_ embeddable via the `exporter` endpoint. Currently this is only used for image rendering
+- **PROPOSAL**: Merge these endpoints together to a unified endpoint for all standalone viewing of content
+
+---
+### Images
+| insights ✅ | dashboards ✅ | person ❌ | cohorts ❌ | groups ❌ |
+|------------|--------------|----------|-----------|----------|
+
+Images are important either for the user (e.g. to include in a powerpoint presentation) or for use by other export methods such as subscriptions
+
+#### Technical considerations
+- [ ] Some things are hard/impossible to render nicely in small sizes (e.g. Funnels).
+- [ ] 
+
+
+---
+### Subscriptions - email:
+| insights ✅ | dashboards ✅ | person ❌ | cohorts ❌ | groups ❌ |
+|------------|--------------|----------|-----------|----------|
+
+Subscriptions via email are essentially Image exports but delivered via email
+
+#### Technical considerations
+- [ ] The exported Image needs to be publicly accessible using some sort of token
+- [ ] Users need to be able to easily unsubscribe without needing to login
+
+---
+### Subscriptions - slack:
+| insights ✅ | dashboards ✅ | person ❌ | cohorts ❌ | groups ❌ |
+|------------|--------------|----------|-----------|----------|
+
+Subscriptions via slack are essentially Image exports but delivered via slack. We may want to consider additional information to be included
+
+#### Technical considerations
+- [ ] The exported Image needs to be publicly accessible using some sort of token
+- [ ] Users need to be able to easily unsubscribe without needing to login
+
+
+---
+### CSV / Excel:
+| insights ✅ | dashboards ✅ | person ✅ | cohorts ✅ | groups ✅ |
+|------------|--------------|----------|-----------|----------|
+
+Resources can be exported directly to a CSV
+
+#### Technical considerations
+- [ ] Exporting 10000s of rows can be costly or even impossible syncronously - it needs to be possible to do this in the background and send to the user when complete
```

Comment:
Will we create one file or several in that case? 

Also note given exporting raw data is costly (esp on cloud) and creates some data control problems - should we have controls/limits around that?

Reply:
It's a good point. Commited that point to the md